vcr/concept/registry.go

```diff
@@ -57,59 +57,34 @@ const (
 	OrganizationCity = "organization.city"
 )
 
-// registry holds parsed concepts which contain all the mappings from concept names to json paths.
+// registry holds parsed credential configs which contain all the mappings from concept names to json paths.
 // Queries are created through the conceptRegistry to add the correct templates.
-// The registry can also do transformations of VCs and queries to the correct format.
-// Concepts are automatically determined from the ConceptTemplates
-// a concept value of <<organization.name>> creates the concept "organization"
+// The registry can also do transformations of VCs to the correct format.
 type registry struct {
-	conceptTemplates map[string][]*Template
-	typedTemplates   map[string]*Template
+	configs []Config
 }
 
 // NewRegistry creates a new registry instance with no templates.
 func NewRegistry() Registry {
 	r := &registry{
-		conceptTemplates: map[string][]*Template{},
-		typedTemplates:   map[string]*Template{},
+		configs: make([]Config, 0),
 	}
 
 	return r
 }
 
-func (r *registry) ConceptTemplates() map[string][]*Template {
-	ct := make(map[string][]*Template, len(r.conceptTemplates))
-
-	// without generics we need to convert the template pointers to Template interface
-	for k, v := range r.conceptTemplates {
-		ts := make([]*Template, len(v))
-		for i, t := range v {
-			ts[i] = t
-		}
-		ct[k] = ts
-	}
-
-	return ct
+func (r *registry) Concepts() []Config {
+	return r.configs
 }
 
-// AddFromString adds a new template to a concept and parses it.
-func (r *registry) Add(conceptTemplate *Template) error {
-	// add to list of templates for same concept name
-	for _, c := range conceptTemplate.rootConcepts() {
-		current, ok := r.conceptTemplates[c]
-		if !ok {
-			current = []*Template{}
-		}
-		current = append(current, conceptTemplate)
-		r.conceptTemplates[c] = current
-	}
-
-	// add to map of specific VC type to template
-	v, ok := conceptTemplate.fixedValues[TypeField]
-	if !ok || v == "" {
+// Add adds a new template to a concept and parses it.
+func (r *registry) Add(config Config) error {
+	// check for type
```

Comment:
good thing this comment is here, otherwise I wouldn't have understood the code ;-)